op-program/host/kvstore/disk.go

```diff
@@ -1,93 +1,68 @@
 package kvstore
 
 import (
-	"encoding/hex"
 	"errors"
 	"fmt"
-	"io"
-	"os"
-	"path"
+	"runtime"
 	"sync"
 
+	"github.com/cockroachdb/pebble"
 	"github.com/ethereum/go-ethereum/common"
 )
 
-// read/write mode for user/group/other, not executable.
-const diskPermission = 0666
-
-// DiskKV is a disk-backed key-value store, every key-value pair is a hex-encoded .txt file, with the value as content.
+// DiskKV is a disk-backed key-value store, with PebbleDB as the underlying DBMS.
 // DiskKV is safe for concurrent use with a single DiskKV instance.
-// DiskKV is safe for concurrent use between different DiskKV instances of the same disk directory as long as the
-// file system supports atomic renames.
 type DiskKV struct {
 	sync.RWMutex
-	path string
+	db *pebble.DB
 }
 
 // NewDiskKV creates a DiskKV that puts/gets pre-images as files in the given directory path.
 // The path must exist, or subsequent Put/Get calls will error when it does not.
 func NewDiskKV(path string) *DiskKV {
-	return &DiskKV{path: path}
-}
+	opts := &pebble.Options{
+		Cache:                    pebble.NewCache(int64(32 * 1024 * 1024)),
+		MaxConcurrentCompactions: runtime.NumCPU,
```

Comment:
Let's disable compactions altogether via `DisableAutomaticCompactions: true` and anything that can cause stall `db.Close`.
The preimage store isn't terribly big and I don't think we'll benefit much with compactions enabled.

Reply:
I think if we're worried about clean shutdowns it would be better to enable sync writes (which I suspect is worth doing - it's a very small performance hit for this kind of use case).  Disabling compactions can cause a full lock up when certain limits are reached and unusual configurations are generally likely to have unexpected side effects like that.

Reply:
I enabled sync writes for pebbledb.  But the default is now the existing file mode so challenger is unaffected by these changes.

Reply:
Thanks for keeping the old one around.